### DISCUSSION TITLE
Unsure if Cherry pick is working

### DISCUSSION THREAD

**Discussion by benfrain** [jesseduffield/lazygit#2314] [Q&A]
I have a commit in branch 'b' that I want to have in branch 'a'.

With 'b' checked out in local branches, I move to the Commits panel and select the commit I want with <kbd>c</kbd> and it goes green around the sha

Then I checkout 'a' and move to the Commits panel. I press <kbd>v</kbd> and get the message "Are you sure you want to cherry-pick the copied commits onto this branch?". I confirm with <kbd>y</kbd> and that message goes, but nothing changes on the file system, and nothing changes in the Commits list of branch 'a'.

First time using cherry-pick. I was expecting to see the commit from branch 'b' now appear in the commit list of branch 'a', along with the file changes that commit introduced.

??

**Comment by jesseduffield** [comment]
I would see if you get the same issue using the git CLI or a separate git gui. I suspect the changes in the commit are already present in the destination branch.

If that's the case, we should really have a popup afterwards that says (who would've thought: those changes were already present!)

**Comment by benfrain** [comment]
Worked from CLI

```
git cherry-pick fdd6f20bf8456ccf657b67f715983c712f66b30d
[homepage 5dccf328] Fix for #249
 Date: Fri Dec 16 10:15:53 2022 +0000
 3 files changed, 27 insertions(+), 46 deletions(-)
```

**Reply by jesseduffield** [reply]
Hmm that's weird: I've never had this happen to me. Are any pre-commit hooks set up (given that they might interfere).

Also, any chance this is a public repo which I could clone to try and reproduce?

**Reply by benfrain** [reply]
> Hmm that's weird: I've never had this happen to me. Are any pre-commit hooks set up (given that they might interfere).

no, we have a single pre-push hook, but nothing pre-commit

I'm afraid it is not a public repo, that would definitely have helped!

**Reply by mark2185** [reply]
Could you maybe reproduce it in a new repo? 

I'm guessing you're cherry picking only one commit, so if it's not too much of a hassle, would you kindly? :)

**Reply by benfrain** [reply]
@mark2185 sure, I'll try and reproduce on a public repo from my home machine later.

**Comment by benfrain** [comment]
Ok, just tried on a public repo and get exactly the same result. Worth mentioning I'm using Lazygit from inside Neovim (not sure why that would matter??)

If you grab this repo: https://github.com/benfrain/rwd4/ and open with Lazygit, you can go into the `testLazyGit` branch and copy  the 41cb4f3ddce5774c6bc6713c227707a0336f7a33 sha (which is a test commit in that branch) and then try and paste that into the commit area of the master branch.

I get exactly the same outcome as described in the initial post.

**Reply by benfrain** [reply]
@mark2185 @jesseduffield if you can replicate let me know and open a bug if needed?

**Reply by jesseduffield** [reply]
Just cloned the repo and tested: on my end the cherry pick worked fine. Here's what my Command Log panel showed:
```
Checkout branch
  git checkout "master"
(Cherry-pick) Paste commits
  Creating TODO file for interactive rebase:

  pick 41cb4f3ddce5774c6bc6713c227707a0336f7a33 Test commit to alt branch

  git rebase --interactive --autostash --keep-empty HEAD
```

**Reply by jesseduffield** [reply]
I would nonetheless open a bug report because it's likely other people are experiencing this bug

**Reply by jesseduffield** [reply]
Interesting that you're using neovim. I've downloaded neovim via homebrew and I've added the neovim lazygit [plugin](https://github.com/kdheepak/lazygit.nvim) via vim-plug.

Using that approach, I also find I'm able to cherry pick fine 🤔
